### PR TITLE
Bug 1434378 - Track startup crashes

### PR DESCRIPTION
**Pull request by akkomar**
This adds `startup_crashes` metric, based on Error Ping's `payload.metadata.StartupCrash`.

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/streaming/ErrorAggregator.scala [hunk 1]
```diff
@@ -247,6 +248,7 @@ object ErrorAggregator {
     stats("count") = Some(1)
     stats("client_id") = ping.meta.clientId
     stats("main_crashes") = Some(1)
+    stats("startup_crashes") = ping.payload.metadata.flatMap(_.get("StartupCrash")).filter(_ == "1").map(_ => 1)
```
[the remark this answers is not in the document]
Comment (akkomar) [reply src/main/scala/com/mozilla/telemetry/streaming/ErrorAggregator.scala]: +1

File: src/main/scala/com/mozilla/telemetry/pings/package.scala [hunk 2]
```diff
@@ -234,12 +232,15 @@ package object pings {
...
   case class CrashPayload(
-      crashDate: String,
-      processType: Option[String],
-      hasCrashEnvironment: Option[Boolean],
-      metadata: Option[Map[String, String]],
-      version: Option[Int])
+                           crashDate: String,
+                           processType: Option[String],
+                           hasCrashEnvironment: Option[Boolean],
+                           metadata: Option[CrashMetadata],
```
Comment (fbertsch) [inline src/main/scala/com/mozilla/telemetry/pings/package.scala]: `metadata` should not be optional here.

File: src/test/scala/com/mozilla/telemetry/streaming/TestErrorAggregator.scala [hunk 3]
```diff
@@ -57,7 +58,8 @@ class TestErrorAggregator extends FlatSpec with Matchers with BeforeAndAfterAll
   "The aggregator" should "sum metrics over a set of dimensions" in {
     import spark.implicits._
     val messages =
-      (TestUtils.generateCrashMessages(k)
+      (TestUtils.generateCrashMessages(k - 1)
+        ++ TestUtils.generateCrashMessages(1, customPayload = Some(""""metadata":{"StartupCrash": "1"}"""))
```
Comment (fbertsch) [inline src/test/scala/com/mozilla/telemetry/streaming/TestErrorAggregator.scala]: Once https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/146 is merged, `metadata` should be added by default to the test crash pings. Good to go for now though

**Comment by akkomar** [comment]
@fbertsch Actually it looks like StartupCrash is documented in accordance with convention for boolean fields: https://bugzilla.mozilla.org/show_bug.cgi?id=1453766

File: src/test/scala/com/mozilla/telemetry/streaming/TestErrorAggregator.scala (hunk 3 above)
Comment (akkomar) [reply src/test/scala/com/mozilla/telemetry/streaming/TestErrorAggregator.scala]: Done.

**Pull request merged**